### ISSUE TITLE
Question: Default .NET Core config and classic .NET Config

### ISSUE THREAD

**Issue by yahorsi** [microsoft/ApplicationInsights-dotnet-server#1247]
Hi Guys

When we add AI to the classic .NET project AI is configured throught the config.
So, we  see it and it is explicit. When we do the same in .NET Core there is no config and when it going on is hidden from us. So, we do not know what telemetry processors/modules/.. is configured.

So, the question is, how explicit .NET AI config compares to the .NET Configuration  when we add it using services.AddApplicationInsightsTelemetry()

Is, default .NET Core configuration described anywhere?

**Comment by cijothomas** [comment]
https://docs.microsoft.com/en-us/azure/azure-monitor/app/asp-net-core#configure-the-application-insights-sdk This is the only doc about the asp.net core sdk. There are some information about the default config spread throughout the doc.
 
It configuration is equivalent of the ai.config based on in asp.net apps. But yes there is no doc listing the default config or comparing default setting for asp.net vs asp.net core.

**Comment by yahorsi** [comment]
> But yes there is no doc listing the default config or comparing default setting for asp.net vs asp.net core.

Exactly, ideally we need to know exact default core config. Having it classic vs core is not that important but will be good to have

**Comment by slipdef** [comment]
I'm not sure what's going with AI actually. MS claims that NET Core is the future (https://devblogs.microsoft.com/dotnet/net-core-is-the-future-of-net/), many first level MS projects drop support for Net Framework (like ASP.NET Core 3 won't support NetFx), but AI config doesn't support Net Core out of the box. So to connect PerfCounters I have to copy some cryptic code from old articles like this one https://apmtips.com/blog/2017/02/13/enable-application-insights-live-metrics-from-code/.

**Comment by cijothomas** [comment]
@slipdef Do you have a specific question about configuring AI on asp.net core, which is not covered here https://docs.microsoft.com/en-us/azure/azure-monitor/app/asp-net-core#configure-the-application-insights-sdk

**Comment by slipdef** [comment]
I'm not interested in ASP.NET Core part specifically, I'm trying to configure .NET Core console app. So I'm looking here https://docs.microsoft.com/en-us/azure/azure-monitor/app/console. At most half of the features supported for file configuration mode (https://docs.microsoft.com/en-us/azure/azure-monitor/app/configuration-with-applicationinsights-config) are described in the docs for console app. Other half suggests me to use that kind of code to setup PerfCounters https://apmtips.com/blog/2017/02/13/enable-application-insights-live-metrics-from-code/. It doesn't look like a configuration code (which must be declarative), it looks like some magic that I copied and shouldn't even touch because I don't understand what is going on there. Actually I don't want to know what does it all mean, all I want to do is smth like: config.EnablePerfCounters = true;

Also I tried to setup exception handling modules and it happened that UnhandledException module is not supported for .net core (https://github.com/microsoft/ApplicationInsights-dotnet-server/issues/1250). Unobserved task exception module is there but I'm not sure if it works or not since the docs missed exceptions section.

**Comment by cijothomas** [comment]
@slipdef You are right. Docs are very light about .net core console apps.
created a tracking issue in docs page: https://github.com/MicrosoftDocs/azure-docs/issues/37389
I cannot comment on when the doc will be updated fully, but if there is a specific configuration you need help with, please respond here.

**Comment by cijothomas** [comment]
@slipdef There is a new doc about new SDK which is best suited to quickly configur entire application insights capabilities with a one line. Please check this: https://docs.microsoft.com/en-us/azure/azure-monitor/app/worker-service I hope it addresses your concerns, atleast partially. please let us know.
